### PR TITLE
CLOUDP-252271 Use gosec linter

### PR DESCRIPTION
**Pull request by slaskawi**
This Pull Request introduces gosec through golangci-lint project. As a side-effect, it removes super-linter integration.

The motivation behind integrating it this way is using a single `//nolint` comment to disable (or mark as false-positive) all linters and not using linter-specific markers (such as `nosec`).

### CODE REVIEW

File: test/e2e/replica_set_arbiter/replica_set_arbiter_test.go [hunk 1]
```diff
@@ -69,9 +69,9 @@ func TestReplicaSetArbiter(t *testing.T) {
 			resourceName:     "mdb4",
 		},
 	}
-	for testName, _ := range tests {
```
Comment (nammn) [inline test/e2e/replica_set_arbiter/replica_set_arbiter_test.go:72]: is the linter complaining over explicit empty assigns?

**Review by nammn — APPROVED** [review]
changes lgtm, but I am interested in the reasons why we have those changes

File: test/e2e/replica_set_operator_upgrade/replica_set_operator_upgrade_test.go [hunk 2]
```diff
@@ -73,7 +73,7 @@ func TestReplicaSetOperatorUpgrade(t *testing.T) {
 // TestReplicaSetOperatorUpgradeFrom0_7_2 is intended to be run locally not in CI.
 // It simulates deploying cluster using community operator 0.7.2 and then upgrading it using newer version.
 func TestReplicaSetOperatorUpgradeFrom0_7_2(t *testing.T) {
-	ctx := context.Background()
+	ctx := context.Background() //nolint
```
[the remark this answers is not in the document]
Comment (slaskawi) [reply test/e2e/replica_set_operator_upgrade/replica_set_operator_upgrade_test.go:76]: The linter though it's not used anywhere. Obviously a false-positive (as it is used a few lines below - it's passed to other methods).

File: test/e2e/replica_set_arbiter/replica_set_arbiter_test.go (hunk 1 above)
Comment (slaskawi) [reply test/e2e/replica_set_arbiter/replica_set_arbiter_test.go:72]: Yeah. it is. I'm not sure why but apparently it doesn't like when you leave the second one empty in a `for`. Kinda makes sense I guess.

**Pull request merged**